src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -651,6 +653,11 @@ private IReadOnlyCollection<ProjectGraphEntryPoint> GetGraphEntryPoints(BuildReq
                 return;
             }
 
+            if (_componentHost.BuildParameters.ReportFileAccesses)
+            {
+                _fileAccessManager.WaitForFileAccessReportCompletion(buildResult.GlobalRequestId, cancellationToken);
```

Comment:
Is GlobalRequestId project-specific? It sounds BuildRequest-specific, in which case I think you might have one project signaling that another is complete, so I figure it's smaller than that.

Reply:
Based on the name, I assumed that `GlobalRequestId` is, well, global. Like unique to the build session.

If there's a better identifier to use, let me know.

Reply:
Oh, I think I see what you're saying. It is intentional that this is BuildRequest specific, and yes that does mean that for a badly-described graph build there could be multiple BuildRequests per project with different sets of targets. However, BuildRequest is 1:1 with BuildResult, which is what is provided, so that should be fine.